### ISSUE TITLE
Uncaught java.lang.StackOverflowError when using Ghidra help with keyboard

### ISSUE THREAD

**Issue by Palacee-hun** [NationalSecurityAgency/ghidra#2686]
**Describe the bug**
The following error pops up after some time when moving the cursor around in the text of a help topic with the arrow keys:
```
java.lang.StackOverflowError
	at java.desktop/java.awt.AWTEventMulticaster.componentMoved(AWTEventMulticaster.java:177)
...
```
and 1023 more of the exact same location. The error text is always identical.
After this error has popped up for the first time in a Ghidra session, two or three presses of the arrow keys are enough to retrigger the error. In this phase arrowing around the help topic tree also manifests this behaviour, which does not do that before the first occurence. After getting a series of such errors, just pressing F1 anywhere to bring up context sensitive help alone is enough to trigger this. This keeps on going till a Ghidra restart.

**To Reproduce**
1. Open Help contents and select a topic that has a longer help text.
2. Focus the text box with the help text in it and try to move around in the text with arrow keys (mainly up and down arrows).
3. When you see the error dialog, just dismiss it, and keep on arrowing. Note how frequent the error has become.
4. Now focus the tree of the help topics and arrow around in that too. Note the same error. You can try this alone before doing these steps in a "fresh" Ghidra session to see that this in itself does not trigger the error.
5. Close the help window, move to some Ghidra controls and bring up context-sensitive help with f1. Most probably this alone will bring up the error dialog. If not try arrowing around in the help text box and you will get the error very soon. No matter how many times you close and relaunch help, the error remains very frequent. Only Ghidra restart "normalises" the situation until the error pops up in the new session.

**Environment (please complete the following information):**
 - OS: Windows 10 64-bit
 - Java Version: AdoptOpenJDK 11.0.9
 - Ghidra Version: 9.2.1

**Comment by dragonmacher** [comment]
We have not seen this before.   Is there any other information in the stack trace besides the line you showed above?

**Comment by Palacee-hun** [comment]
No, nothing else, just 1024 instances of this line in total (I suppose that is the stack depth in this respect and there's some weird recursion going on here), plus some system info which I put in my starting comment.

**Comment by dragonmacher** [comment]
This observation will likely not help you, but I can see a path for this to happen.  

TL;DR: how much RAM is you Ghidra getting?   (You can see this from the Front End by  cliking Help -> Show VM Memory).

In Java's Help UI code, they add a listener every time a page is loaded into the viewer.   The code will remove that listener as well.   The issue, I think, is that the UI code uses a `WeakReference` to know how to remove the listener.   If that reference has been garbage collected, then the listener will not get removed.   Eventually, if the listener is added more times than the stack can handled, the stack overflow will occur.

What is not clear is why nobody else is running into this issue.  While debugging, my weak reference is not being garbage collected.  This implies that my system has enough RAM that the garbage collector does not feel the need to reclaim the memory.   Or, another object is holding onto the reference, keeping it from getting collected.

**Comment by Palacee-hun** [comment]
My VM stats and behaviour are the following:
Case 1.: 4 Gig system, just Windows ui is running, more than 65 % of RAM is free, Ghidra is loaded with a simple project with a 80 Kb portable executable in it
Max memory: 987,758 K
Total memory: 75, 497 K (this is very strange, why this is so low compared to max mem, when there's ample free RAM ???)
Free memory: this oscillates between around 40 Mb and 17 Mb during a cycle that last approx. three minutes, it comes down from 40 Mb gradually and smoothly to 17 Mb without me touching the system, and then jumps back to 40 Mb and starts over
Used memory: always total memory - free memory of course
Case 2.: same system, same Ghidra project, but file manager and browser with approx. 20 tabs are open, but cca. 50 % of the 4 Gb RAM is still free
exact same stats and behaviour
My system is very snappy and responsive all the time, Ghidra startup time is under 5 secs, browser (Edge) startup time is under 2 secs with 20 tabs open
Max memory is all right as far as I know, that defaults to 1 Gb on a 64-bit system (maxmem environment variable) and that is so. Total memory is striking in multiple ways for me.

**Comment by dragonmacher** [comment]
> Total memory: 75, 497 K (this is very strange, why this is so low compared to max mem, when there's ample free RAM ???)

This is normal in that the VM will only use what it needs

> Free memory: this oscillates between around 40 Mb and 17 Mb during a cycle that last approx. three minutes, it comes down from 40 Mb gradually and smoothly to 17 Mb without me touching the system, and then jumps back to 40 Mb and starts over
Used memory: always total memory - free memory of course

This sounds very strange to me if you are not actively working in the tool.   This may be why that weak reference I mentioned is going away.  I usually witness the VM not reclaiming memory until it is needed.  Usually a stasis is reached.

> Ghidra startup time is under 5 secs,

Wow, that is pretty good.

If this issue is a major issue for your workflow, then I would suggest trying newer versions of Java.    However, given the code that has the issue, I'm not sure it would help.    **The more drastic solution is to copy and manually edit Java's code to workaround the issue.  Then use that code along with your Ghidra to override Java's issue.  This is a bit advanced to do.**
